### PR TITLE
Jax windows restrictions

### PR DESCRIPTION
**Pull request by kratman**
# Description

Restrictions on testing and installing Jax can be removed after #3550

Related: #3941

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [x] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by kratman** [comment]
@agriyakhetarpal This should be all of the jax stuff. Since I am waiting on #3954 I will do the coverage fixes in a third PR

**Review by agriyakhetarpal — DISMISSED** [review]
Thanks!

**Comment by kratman** [comment]
Sorry pushed another change after your review

File: docs/source/user_guide/installation/install-from-source.rst [hunk 1]
```diff
@@ -46,7 +46,7 @@ You can install the above with
 
 	.. code:: bash
 
-		brew install python openblas gcc gfortran graphviz libomp cmake
+		brew install python graphviz libomp cmake
```
Comment (agriyakhetarpal) [inline docs/source/user_guide/installation/install-from-source.rst:49]: Actually, we need these here, can you revert this change? These are pre-installed on macOS GitHub runners, but users might not have them.

Comment (kratman) [reply docs/source/user_guide/installation/install-from-source.rst:49]:
```suggestion
		brew install python openblas gcc gfortran graphviz libomp cmake
```

Comment (kratman) [reply docs/source/user_guide/installation/install-from-source.rst]: Done

**Review by agriyakhetarpal — APPROVED** [review]

**Comment by kratman** [comment]
Coverage is still failing in #3954, so it will fail here too

**Review by agriyakhetarpal — DISMISSED** [review]
I updated the branch, it should be good to go now. Please feel free to merge

**Comment by agriyakhetarpal** [comment]
Based on the failing tests, I think the long-standing issue of Homebrew's `gcc` not coming with `gfortran` is still not fixed. We should add them back again because we wouldn't want workflows like `publish_pypi.yml` to fail for unrelated reasons on releases.

File: .github/workflows/publish_pypi.yml [hunk 2]
```diff
@@ -116,8 +116,7 @@ jobs:
       - name: Install SuiteSparse and SUNDIALS on macOS
         if: matrix.os == 'macos-12'
         run: |
-          brew install graphviz openblas libomp
-          brew reinstall gcc
+          brew install graphviz libomp
```
Comment (kratman) [inline .github/workflows/publish_pypi.yml:119]:
```suggestion
          brew install graphviz libomp gcc
```

File: .github/workflows/publish_pypi.yml [hunk 3]
```diff
@@ -173,8 +172,7 @@ jobs:
 
       - name: Install SuiteSparse and SUNDIALS on macOS
         run: |
-          brew install graphviz openblas libomp
-          brew reinstall gcc
+          brew install graphviz libomp
```
Comment (kratman) [inline .github/workflows/publish_pypi.yml:175]:
```suggestion
          brew install graphviz libomp gcc
```

File: .github/workflows/run_periodic_tests.yml [hunk 4]
```diff
@@ -77,8 +77,7 @@ jobs:
         if: matrix.os == 'macos-12' || matrix.os == 'macos-14'
         run: |
           brew analytics off
-          brew install graphviz openblas libomp
-          brew reinstall gcc
+          brew install graphviz libomp
```
Comment (kratman) [inline .github/workflows/run_periodic_tests.yml:80]:
```suggestion
          brew install graphviz libomp gcc
```

File: .github/workflows/test_on_push.yml [hunk 5]
```diff
@@ -82,8 +82,7 @@ jobs:
         # sometimes gfortran cannot be found, so reinstall gcc just to be sure
         run: |
           brew analytics off
-          brew install graphviz openblas libomp
-          brew reinstall gcc
+          brew install graphviz libomp
```
Comment (kratman) [inline .github/workflows/test_on_push.yml:85]:
```suggestion
          brew install graphviz libomp gcc
```

File: .github/workflows/test_on_push.yml [hunk 6]
```diff
@@ -226,8 +225,7 @@ jobs:
...
```
Comment (kratman) [inline .github/workflows/test_on_push.yml:228]:
```suggestion
          brew install graphviz libomp gcc
```

File: .github/workflows/publish_pypi.yml [hunk 7]
```diff
@@ -116,8 +116,7 @@ jobs:
       - name: Install SuiteSparse and SUNDIALS on macOS
         if: matrix.os == 'macos-12'
         run: |
-          brew install graphviz openblas libomp
-          brew reinstall gcc
+          brew install graphviz libomp gcc
```
Comment (kratman) [inline .github/workflows/publish_pypi.yml:119]:
```suggestion
          brew install graphviz libomp
          brew reinstall gcc
```

File: .github/workflows/publish_pypi.yml [hunk 8]
```diff
@@ -173,8 +172,7 @@ jobs:
 
       - name: Install SuiteSparse and SUNDIALS on macOS
         run: |
-          brew install graphviz openblas libomp
-          brew reinstall gcc
+          brew install graphviz libomp gcc
```
Comment (kratman) [inline .github/workflows/publish_pypi.yml:175]:
```suggestion
          brew install graphviz libomp 
          brew reinstall gcc
```

File: .github/workflows/run_periodic_tests.yml [hunk 9]
```diff
@@ -77,8 +77,7 @@ jobs:
         if: matrix.os == 'macos-12' || matrix.os == 'macos-14'
         run: |
           brew analytics off
-          brew install graphviz openblas libomp
-          brew reinstall gcc
+          brew install graphviz libomp gcc
```
Comment (kratman) [inline .github/workflows/run_periodic_tests.yml:80]:
```suggestion
          brew install graphviz libomp  
          brew reinstall gcc
```

File: .github/workflows/test_on_push.yml [hunk 10]
```diff
@@ -82,8 +82,7 @@ jobs:
         # sometimes gfortran cannot be found, so reinstall gcc just to be sure
         run: |
           brew analytics off
-          brew install graphviz openblas libomp
-          brew reinstall gcc
+          brew install graphviz libomp gcc
```
Comment (kratman) [inline .github/workflows/test_on_push.yml:85]:
```suggestion
          brew install graphviz libomp  
          brew reinstall gcc
```

File: .github/workflows/test_on_push.yml [hunk 11]
```diff
@@ -226,8 +225,7 @@ jobs:
...
```
Comment (kratman) [inline .github/workflows/test_on_push.yml:228]:
```suggestion
          brew install graphviz libomp  
          brew reinstall gcc
```

File: .github/workflows/publish_pypi.yml [hunk 12]
```diff
@@ -116,7 +116,7 @@ jobs:
       - name: Install SuiteSparse and SUNDIALS on macOS
         if: matrix.os == 'macos-12'
         run: |
-          brew install graphviz openblas libomp
+          brew install graphviz libomp
```
Comment (kratman) [inline .github/workflows/publish_pypi.yml:119]:
```suggestion
          brew install graphviz libomp
```

File: .github/workflows/publish_pypi.yml [hunk 13]
```diff
@@ -173,7 +173,7 @@ jobs:
 
       - name: Install SuiteSparse and SUNDIALS on macOS
         run: |
-          brew install graphviz openblas libomp
+          brew install graphviz libomp
```
Comment (kratman) [inline .github/workflows/publish_pypi.yml:176]:
```suggestion
          brew install graphviz libomp
```

File: .github/workflows/run_periodic_tests.yml [hunk 14]
```diff
@@ -77,7 +77,7 @@ jobs:
         if: matrix.os == 'macos-12' || matrix.os == 'macos-14'
         run: |
           brew analytics off
-          brew install graphviz openblas libomp
+          brew install graphviz libomp
```
Comment (kratman) [inline .github/workflows/run_periodic_tests.yml:80]:
```suggestion
          brew install graphviz libomp
```

File: .github/workflows/test_on_push.yml [hunk 15]
```diff
@@ -82,7 +82,7 @@ jobs:
         # sometimes gfortran cannot be found, so reinstall gcc just to be sure
         run: |
           brew analytics off
-          brew install graphviz openblas libomp
+          brew install graphviz libomp
```
Comment (kratman) [inline .github/workflows/test_on_push.yml:85]:
```suggestion
          brew install graphviz libomp
```

File: .github/workflows/test_on_push.yml [hunk 16]
```diff
@@ -226,7 +226,7 @@ jobs:
...
```
Comment (kratman) [inline .github/workflows/test_on_push.yml:229]:
```suggestion
          brew install graphviz libomp
```

**Review by agriyakhetarpal — APPROVED** [review]
Had to trigger a re-run for some, but all tests pass now, we should be good to go. Thanks!
